ui/src/common/components/CitationSummaryGraph/CitationSummaryGraph.jsx

```diff
@@ -0,0 +1,130 @@
+  YAxis,
+  HorizontalGridLines,
+  VerticalBarSeries,
+  LabelSeries,
+  FlexibleWidthXYPlot,
+  DiscreteColorLegend,
+} from 'react-vis';
+import PropTypes from 'prop-types';
+import { Map } from 'immutable';
+import { Row, Col } from 'antd';
+import './CitationSummaryGraph.scss';
+import styleVariables from '../../../styleVariables';
+
+class CitationSummaryGraph extends Component {
+  LEGEND_ITEMS = [
+    { title: 'Citeable', color: styleVariables['primary-color'] },
+    { title: 'Published', color: styleVariables['orange-6'] },
+  ];
+
+  keyToLabel = new Map({
+    '0.0-1.0': '0',
+    '1.0-10.0': '1-9',
+    '10.0-50.0': '10-49',
+    '50.0-100.0': '50-99',
+    '100.0-250.0': '100-249',
+    '250.0-500.0': '250-499',
+    '500.0-*': '500+',
+  });
+
+  static getSeriesData(data) {
+    const result = [];
+    data.forEach(bucket => {
+      const docCount = bucket.get('doc_count');
+      const countLabel = docCount !== 0 ? docCount.toString() : null;
+      result.push({ x: bucket.get('key'), y: docCount, label: countLabel });
+    });
+    return result;
+  }
+
+  render() {
+    const { citationSummary } = this.props;
+    const citeableData = citationSummary.getIn([
+      'citations',
+      'buckets',
+      'all',
+      'citation_buckets',
+      'buckets',
+    ]);
+    const publishedData = citationSummary.getIn([
+      'citations',
+      'buckets',
+      'published',
+      'citation_buckets',
+      'buckets',
+    ]);
+    const publishedSeriesData = CitationSummaryGraph.getSeriesData(
+      publishedData
+    );
+    const citeableSeriesData = CitationSummaryGraph.getSeriesData(citeableData);
+
+    const yDomainMax = Math.max(
+      ...publishedSeriesData.map(d => d.y),
+      ...citeableSeriesData.map(d => d.y)
+    );
+    const barWidth = 0.5;
+    return (
+      <div className="__CitationSummaryGraph__">
+        <Fragment>
+          <Row type="flex" align="middle">
+            <Col span={1}>
+              <div className="axis-label y-axis">Papers</div>
+            </Col>
+            <Col span={23}>
+              <FlexibleWidthXYPlot
+                height={300}
+                xType="ordinal"
+                animation
+                yDomain={[0, yDomainMax * 1.3]}
+              >
+                <HorizontalGridLines />
+                <XAxis
+                  className="xAxis"
+                  tickFormat={v => this.keyToLabel.get(v)}
+                />
+                <YAxis />
+                <VerticalBarSeries
+                  data={citeableSeriesData}
+                  color={styleVariables['primary-color']}
+                  barWidth={barWidth}
+                />
+                <LabelSeries
+                  data={citeableSeriesData}
+                  labelAnchorX="end"
+                  labelAnchorY="text-after-edge"
+                />
+                <VerticalBarSeries
+                  data={publishedSeriesData}
+                  color={styleVariables['orange-6']}
```

Comment:
why don't you put those in a constant called `ORANGE` and `BLUE` and use it everywhere.